rust/src/dns/parser.rs

```diff
@@ -141,6 +141,19 @@ fn dns_parse_answer<'a>(
                         1
                     }
                 };
+                // edge case for additional section of type=OPT
+                // with empty data (data length = 0x0000)
+                if val.data.is_empty() && val.rrtype == DNS_RECORD_TYPE_OPT {
+                    answers.push(DNSAnswerEntry {
+                        name: val.name.clone(),
+                        rrtype: val.rrtype,
+                        rrclass: val.rrclass,
+                        ttl: val.ttl,
+                        data: DNSRData::OPT(Vec::new()),
+                    });
+                    input = rem;
+                    continue;
+                }
```

Comment:
Is there any value in logging this? I'm looking at a PCAP that hits the case, and what is logged bears no resemblance to what is logged, so it feels like extra noise.
![2024-05-31T12:31:26,703297294-06:00](https://github.com/OISF/suricata/assets/644115/e6848208-a93f-4b0e-8cef-e879510ad11e)

Reply:
I agree, there is likely little value in logging the OPT records with no rdata bytes. But this logic could be added to `log.rs` to make sure these aren't logged instead of skipping the record here. This PR doesn't actually include any custom logging logic to handle additional records of type OPT, so right now the rdata will never be logged. Would you prefer for me to implement this and add it to the PR (similar to how PTR, SOA, SSHFP and SRV are handled) or simply not log OPT at all for now?

I think it is important to keep the structure of the `DNSTransaction` true to the parsed packet (i.e. all dns records present, even these OPT ones). One reason for this is I have another feature (ticket not yet created) for an action called "redirect". It relies on the whole packet being parsed with the record stored accurately in the `DNSTransaction`. Redirect provides the ability to inject a custom DNS response with modified A/AAAA records (some vendors call the action DNS sinkhole).